File: brain/92652e51-9e6d-47d5-82b4-3381722aecf2/scratch/read_dimensions.js
const fs = require('fs');

function getJpegSize(filePath) {
  const buffer = fs.readFileSync(filePath);
  let i = 4;
  while (i < buffer.length) {
    const marker = buffer.readUInt16BE(i);
    i += 2;
    if (marker === 0xFFC0 || marker === 0xFFC2) {
      i += 2; // skip length
      const precision = buffer[i++];
      const height = buffer.readUInt16BE(i);
      i += 2;
      const width = buffer.readUInt16BE(i);
      return { width, height };
    } else {
      const length = buffer.readUInt16BE(i);
      i += length;
    }
  }
  return null;
}

try {
  console.log('Arabescato:', getJpegSize('public/images/Links/Arabescato Vagli Face 1_1.jpg'));
  console.log('Oyster Face 1:', getJpegSize('public/images/Links/Calacatta Oyster Face 1.jpg'));
  console.log('Oyster Bookmatch:', getJpegSize('public/images/Calacatta Oyster/Bookmatch.jpg'));
} catch (e) {
  console.error(e);
}

File: brain/92652e51-9e6d-47d5-82b4-3381722aecf2/scratch/read_dimensions.py
from PIL import Image
import os

images = [
    'public/images/Links/Arabescato Vagli Face 1_1.jpg',
    'public/images/Links/Calacatta Oyster Face 1.jpg',
    'public/images/Calacatta Oyster/Bookmatch.jpg'
]

for img_path in images:
    if os.path.exists(img_path):
        try:
            with Image.open(img_path) as img:
                print(img_path, img.size)
        except Exception as e:
            print(img_path, "Error:", e)
    else:
        print(img_path, "does not exist")
